File: back/src/main/java/com/back/domain/schedule/dto/ShowScheduleListResponse.java
package com.back.domain.schedule.dto;

import com.back.domain.schedule.entity.Schedule;
import java.time.LocalDateTime;

public record ShowScheduleListResponse(
        Long scheduleId,
        int round,
        LocalDateTime scheduleDate,
        long remainingSeats
) {
    public static ShowScheduleListResponse of(Schedule schedule, long remainingSeats) {
        return new ShowScheduleListResponse(
                schedule.getScheduleId(),
                schedule.getRound(),
                schedule.getScheduleDate(),
                remainingSeats
        );
    }
}

File: back/src/main/kotlin/com/back/domain/schedule/dto/ShowScheduleListResponse.kt
package com.back.domain.schedule.dto

import com.back.domain.schedule.entity.Schedule
import java.time.LocalDateTime

data class ShowScheduleListResponse(
    val scheduleId: Long?,
    val round: Int,
    val scheduleDate: LocalDateTime,
    val remainingSeats: Long
) {
    companion object {
        @JvmStatic
        fun of(schedule: Schedule, remainingSeats: Long): ShowScheduleListResponse =
            ShowScheduleListResponse(
                scheduleId = schedule.scheduleId,
                round = schedule.round,
                scheduleDate = schedule.scheduleDate,
                remainingSeats = remainingSeats
            )
    }
}
